website/source/docs/guides/connect-services.html.md

```diff
@@ -0,0 +1,444 @@
+local Connect Proxy at port 5000. This traffic then traverses the Connect mesh
+over dynamic ports. The traffic exits the Connect mesh from the counting service's
+local proxy. The proxy sends this traffic to the counting service itself at port
+9003.][img-flow]][img-flow]
+
+While this guide uses elements that are not suitable for production
+environments—Consul dev agents, internal proxies, and mock services—it will
+teach you the common process for deploying your own services using Consul
+Connect. At the end of this guide, we also present additional information
+about adapting this process to more production-like environments.
+
+## Prerequisites
+
+To complete this guide, you will need a local [dev agent], which enables
+Connect by default.
+
+This guide uses the following example service applications. Download and unzip
+the executables to follow along.
+
+- [Counting Service]
+- [Dashboard Service]
+
+### Verify your Consul agent health
+
+To ensure that Consul is running and accessible from the command line, use the
+`consul members` command to verify your agent status.
+
+```shell
+$ consul members
+Node            Address         Status  Type    Build  Protocol  DC   Segment
+hostname.local  127.0.0.1:8301  alive   server  1.6.1  2         dc1  <all>
+```
+
+If you receive an error message, verify that you have a local Consul dev agent
+running and try again.
+
+## Register the services and sidecar proxies
+
+Services have to be registered with Consul. Consul shares this information
+around the cluster so that operators or other services can determine the
+location of a service. Connect also uses service registrations to determine
+where to send proxied traffic to.
+
+There are several ways to register services in Consul:
+
+- directly from a Consul-aware application
+
+- from an orchestrator, like [Nomad][services-nomad] or [Kubernetes][services-k8s]
+
+- [using configuration files][services-config] that are loaded at node startup
+
+- [using the API][services-api] to register them with a JSON or HCL
+  specification
+
+- [using the CLI][services-cli] to simplify this submission process
+
+For this guide, we will use the [`consul service register`][services-cli] CLI
+command to load them into the catalog.
+
+### Create the counting service definition
+
+First, define the Counting service and its sidecar proxy in a file named
+`counting.hcl`. The definition should include the name of the service, the port
+the service listens on, and a [connect] block with the [sidecar_service] block.
+This block is empty so Consul will use default parameters. The definition also
+includes an optional service health check.
+
+```hcl
+service {
+  name = "counting"
+  id = "counting-1"
+  port = 9003
+
+  connect {
+    sidecar_service {}
+  }
+
+  check {
+    id       = "counting-check"
+    http     = "http://localhost:9003/health"
+    method   = "GET"
+    interval = "1s"
+    timeout  = "1s"
```

Comment:
Only one of these values may be present on a check definition, not both. See https://github.com/hashicorp/consul/pull/6753.